src/adapters/RoomsJSONAdapter.js

```diff
@@ -0,0 +1,83 @@
+import {Observable} from 'rxjs';
+
+import RoomsAdapter from './RoomsAdapter';
+
+/**
+ * @typedef RoomJSON
+ * @param {object} datasource An object that contains a set of rooms keyed by ID.
+ * @example
+ * {
+ *  "room-1": {
+ *     "ID": "room-1",
+ *     "title": "title",
+ *     "roomType": "group"
+ *  }
+ * }
+ */
+
+/*
+ * Implements the RoomsAdapter interface with a JSON object as its datasource. See @RoomsJSON
+ */
+export default class RoomsJSONAdapter extends RoomsAdapter {
+  constructor(datasource) {
+    super(datasource);
+    this.getRoom = this.getRoom.bind(this);
```

Comment:
Is there a reason for not binding `getPreviousRoomActivities` and `getRoomActivities`?